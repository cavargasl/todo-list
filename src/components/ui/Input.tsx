import { cn } from "@/utils"

type InputProps = React.InputHTMLAttributes<HTMLInputElement>

export default function Input({ className, ...props }: InputProps) {
  return (
    <input
      className={cn(
        "w-full rounded-md border bg-black px-2 outline-none focus:border-rose-500",
        className
      )}
      {...props}
    />
  )
}
