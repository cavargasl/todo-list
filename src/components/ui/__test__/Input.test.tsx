import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Input from "../Input";

describe("Input component", () => {
  test("renders correctly", () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  test("handles onChange event correctly", async () => {
    render(<Input placeholder="Typing here" />);
    const inputElement = screen.getByPlaceholderText("Typing here");
    await userEvent.type(inputElement, "New Task");
    expect(inputElement).toHaveValue("New Task");
  });

  test("renders with custom className and disabled", () => {
    render(<Input className="custom-class" role="input" disabled />);
    const inputElement = screen.getByRole("input");
    expect(inputElement).toHaveClass("custom-class");
    expect(inputElement).toBeDisabled();
  });
});
