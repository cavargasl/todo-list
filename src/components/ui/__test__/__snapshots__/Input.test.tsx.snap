// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > renders correctly 1`] = `
<div>
  <input
    class="w-full rounded-md border bg-black px-2 outline-none focus:border-rose-500"
  />
</div>
`;
